tests/io_test.py

```diff
@@ -876,6 +876,278 @@ def test_dataframe_segarr(self):
+            a.to_hdf(f"{tmp_dirname}/str_test", dataset="test_set")
+            b.to_hdf(f"{tmp_dirname}/str_test", mode="append")
+            f_list = glob.glob(f"{tmp_dirname}/str_test_*")
+            orig_size = 0
+            for f in f_list:
+                orig_size += os.path.getsize(f)
+            c.update_hdf(f"{tmp_dirname}/str_test", dataset="test_set")
+
+            new_size = 0
+            for f in f_list:
+                new_size += os.path.getsize(f)
+            # ensure that the column was actually overwritten
+            self.assertLessEqual(new_size, orig_size)
+            data = ak.read_hdf(f"{tmp_dirname}/str_test_*")
+            self.assertListEqual(data["test_set"].to_list(), c.to_list())
+
+        with tempfile.TemporaryDirectory(dir=IOTest.io_test_dir) as tmp_dirname:
+            a.to_hdf(f"{tmp_dirname}/str_test", dataset="test_set")
+            b.to_hdf(f"{tmp_dirname}/str_test", mode="append")
+            f_list = glob.glob(f"{tmp_dirname}/str_test_*")
+            orig_size = 0
+            for f in f_list:
+                orig_size += os.path.getsize(f)
+            # hdf5 only releases memory if overwritting last dset so overwrite first
+            c.update_hdf(f"{tmp_dirname}/str_test", dataset="test_set", repack=False)
+
+            new_size = 0
+            for f in f_list:
+                new_size += os.path.getsize(f)
+            # ensure that the column was actually overwritten
+            self.assertGreaterEqual(new_size, orig_size)
+            data = ak.read_hdf(f"{tmp_dirname}/str_test_*")
+            self.assertListEqual(data["test_set"].to_list(), c.to_list())
+
+    def test_hdf_overwrite_dataframe(self):
+        df = ak.DataFrame({
+            "a": ak.arange(1000),
+            "b": ak.random_strings_uniform(0, 16, 1000),
+            "c": ak.arange(1000, dtype=bool),
+            "d": ak.randint(0, 50, 1000)
+        })
+        odf = ak.DataFrame({
+            "b": ak.randint(0, 25, 50),
+            "c": ak.arange(50, dtype=bool),
+        })
+        with tempfile.TemporaryDirectory(dir=IOTest.io_test_dir) as tmp_dirname:
+            df.to_hdf(f"{tmp_dirname}/df_test")
+            f_list = glob.glob(f"{tmp_dirname}/df_test_*")
+            orig_size = 0
+            for f in f_list:
+                orig_size += os.path.getsize(f)
+            # hdf5 only releases memory if overwritting last dset so overwrite first
+            odf.update_hdf(f"{tmp_dirname}/df_test")
+
+            new_size = 0
+            for f in f_list:
+                new_size += os.path.getsize(f)
+            # ensure that the column was actually overwritten
+            self.assertLessEqual(new_size, orig_size)
+            data = ak.read_hdf(f"{tmp_dirname}/df_test_*")
+            self.assertListEqual(data["b"].to_list(), odf["b"].to_list())
+            self.assertListEqual(data["c"].to_list(), odf["c"].to_list())
```

Comment:
im not sure we need to update the test but i wanted to make sure, `data["a/d"]` still matches `df["a/d"]` right?